From d5d01c6336ba2067fd1273bae3ad5bd713ea9a4b Mon Sep 17 00:00:00 2001
From: Tony Lin <tony.lin@freescale.com>
Date: Tue, 21 Jun 2011 13:29:58 +0800
Subject: [PATCH 585/670] ENGR00142099-3 [MX233/MX28]implement fastpath in
 driver level

mmc driver will choose to use pio mode or interrupt mode by parameter
passed from board layer.

Signed-off-by: Tony Lin <tony.lin@freescale.com>
---
 drivers/mmc/host/mxs-mmc.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/mmc/host/mxs-mmc.c b/drivers/mmc/host/mxs-mmc.c
index b5bca26..8f9367f 100644
--- a/drivers/mmc/host/mxs-mmc.c
+++ b/drivers/mmc/host/mxs-mmc.c
@@ -141,7 +141,7 @@ struct mxs_mmc_host {
 
 	spinlock_t lock;
 	int sdio_irq_en;
-	int pio_size;
+	int fastpath_sz;
 };
 
 /* Return read only state of card */
@@ -816,7 +816,7 @@ static void mxs_mmc_start_cmd(struct mxs_mmc_host *host,
 		break;
 	case MMC_CMD_ADTC:
 		data_size = host->cmd->data->blksz * host->cmd->data->blocks;
-		mxs_mmc_adtc(host, data_size < host->pio_size);
+		mxs_mmc_adtc(host, data_size < host->fastpath_sz);
 		break;
 	default:
 		dev_warn(host->dev, "Unknown MMC command\n");
@@ -1229,7 +1229,7 @@ static int __init mxs_mmc_probe(struct platform_device *pdev)
 	host->dev = dev;
 
 	host->sdio_irq_en = 0;
-	host->pio_size = 0;
+	host->fastpath_sz = mmc_data->fastpath_sz;
 	/* Set minimal clock rate */
 	host->clk = clk_get(dev, mmc_data->clock_mmc);
 	if (IS_ERR(host->clk)) {
-- 
1.8.0

